### PR TITLE
CTOOLS-34: Add file access token (ported from lusidjam)

### PR DESCRIPTION
**Pull request by adamMQM**
Add file access token config loader
Add file access token config loader to default config loaders

# Pull Request Checklist

- [ ] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [ ] Tests pass
- [ ] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

**Review by Matt-Storey-Finbourne — APPROVED** [review]

File: generate/templates/extensions/file_access_token.mustache [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Acts as a string so can be concatenated to auth headers
+    """
+
+    def __init__(self, access_token_location: str, expiry_time:int = 120):
+        if access_token_location is None or access_token_location == "":
+            raise ValueError("access_token_location must be a non-empty string")
+        self.__access_token_location = access_token_location
+        self.__expiry_time = expiry_time 
+        self.__token_data = {
+            "expires": datetime.datetime.now(),
```
Comment (hanialsibai) [inline generate/templates/extensions/file_access_token.mustache:19]: I realise this must be from lusid_jam, but it seems odd to default this one to `datetime.datetime.now()` as that time doesn't seem to be used

Would it be clearer for init here to set to `None` and add an `or self.__token_data["expires"] is None` to line 32?

File: generate/templates/extensions/configuration_loaders.mustache [hunk 2]
```diff
@@ -141,10 +142,34 @@ class EnvironmentVariablesConfigurationLoader:
...
+        client_id:Optional[str]=None,
+        client_secret:Optional[str]=None,
+        app_name:Optional[str]=None,
+        certificate_filename:Optional[str]=None,
+        proxy_address:Optional[str]=None,
+        proxy_username:Optional[str]=None,
+        proxy_password:Optional[str]=None,
+        access_token:Optional[str]=None
+        ):
         """kwargs passed to this constructor used to build ApiConfiguration
```
Comment (hanialsibai) [inline generate/templates/extensions/configuration_loaders.mustache:159]: This comment might need updating or removed

File: generate/templates/extensions/configuration_loaders.mustache [hunk 3]
```diff
@@ -155,13 +180,51 @@ class ArgsConfigurationLoader:
...
+            "proxy_username" : self.__proxy_username, 
+            "proxy_password" : self.__proxy_password, 
+            "access_token" : self.__access_token
+        }
+
+
+class FileTokenConfigurationLoader:
+    """ConfigurationLoader which loads in access token from file
+        if FBN_ACCESS_TOKEN_FILE is set,
+        or if an access_token_location is passed to the initialiser
```
Comment (hanialsibai) [inline generate/templates/extensions/configuration_loaders.mustache:202]:
```
ConfigurationLoader which loads in access token from file location
specified in environment variable FBN_ACCESS_TOKEN_FILE
        or if an access_token_location is passed to the initialiser
 ```
Does that seem more explicit/clearer?

File: generate/templates/extensions/configuration_loaders.mustache [hunk 4]
```diff
@@ -141,10 +142,34 @@ class EnvironmentVariablesConfigurationLoader:
...
+        self.__client_secret = client_secret
+        self.__app_name = app_name
+        self.__certificate_filename = certificate_filename
+        self.__proxy_address = proxy_address
+        self.__proxy_username = proxy_username
+        self.__proxy_password = proxy_password
+        self.__access_token = access_token
 
     def load_config(self) -> Dict[str, str]:
         """load configuration from kwargs passed to constructor
```
Comment (hanialsibai) [inline generate/templates/extensions/configuration_loaders.mustache:175]: Comment should be updated here. 

I'm not sure which one is the most visible, but would be good for one of these comments to have a description + example of each arg

**Pull request merged**